lib/graphql/resultPage.ts

```diff
@@ -1,22 +1,33 @@
 import { PageInfo } from "./schema";
 import { IFetch } from "./interfaces";
+import { FetchOptions } from "./types";
 
-export class ResultPage<T> {
-  items: Array<T>;
+export class ResultPage<ModelType, FetchOptionsType = FetchOptions> {
+  items: Array<ModelType>;
   pageInfo: PageInfo;
-  nextPage?: () => Promise<ResultPage<T>>;
-  previousPage?: () => Promise<ResultPage<T>>;
+  nextPage?: () => Promise<ResultPage<ModelType, FetchOptionsType>>;
+  previousPage?: () => Promise<ResultPage<ModelType, FetchOptionsType>>;
 
-  constructor(model: IFetch<T>, items: Array<T>, pageInfo: PageInfo) {
+  constructor(
+    model: IFetch<ModelType, FetchOptionsType>,
+    items: Array<ModelType>,
+    pageInfo: PageInfo,
+    args?: FetchOptionsType
+  ) {
     this.items = items;
     this.pageInfo = pageInfo;
 
     if (pageInfo.hasNextPage) {
-      this.nextPage = () => model.fetch({ after: pageInfo.endCursor });
+      this.nextPage = () =>
+        model.fetch({ ...args, after: pageInfo.endCursor } as FetchOptionsType);
     }
 
     if (pageInfo.hasPreviousPage) {
-      this.previousPage = () => model.fetch({ before: pageInfo.startCursor });
+      this.previousPage = () =>
+        model.fetch({
+          ...args,
+          before: pageInfo.startCursor
+        } as FetchOptionsType);
```

Comment:
nit: can we skip this `as FetchOptionsType` here and at two other places? `args?` must not be optional for that to satisfy `tsc`.